sklearn/tests/metadata_routing_common.py

```diff
@@ -25,55 +26,69 @@
 from sklearn.utils.multiclass import _check_partial_fit_first_call
 
 
-def record_metadata(obj, method, record_default=True, **kwargs):
-    """Utility function to store passed metadata to a method.
+def record_metadata(obj, record_default=True, **kwargs):
+    """Utility function to store passed metadata to a method of obj.
 
     If record_default is False, kwargs whose values are "default" are skipped.
     This is so that checks on keyword arguments whose default was not changed
     are skipped.
 
     """
+    stack = inspect.stack()
+    method = stack[1].function
+    parent = stack[2].function
```

Comment:
I'm not sure that many people are aware of the `stack` method or how does it work.

Do we want to use the calle/caller named

```python
callee_method = stack[1].function
caller_method = stack[2].function
```

Reply:
> I'm not sure that many people are aware of the stack method or how does it work.

I'm not sure what to take from that 😁 

But sure, I can rename the variables.

Reply:
Sorry, I was probably tired when I was writing. I just wanted to say that renaming the variables might reduce the need to have to go and check that the `inspect.stack` is doing :)